eth/stagedsync/stage_bodies.go

```diff
@@ -117,27 +121,59 @@ func BodiesForward(
 	var prevDeliveredCount float64 = 0
 	var prevWastedCount float64 = 0
 	timer := time.NewTimer(1 * time.Second) // Check periodically even in the abseence of incoming messages
-	var blockNum uint64
 	var req *bodydownload.BodyRequest
 	var peer [64]byte
 	var sentToPeer bool
 	stopped := false
 	prevProgress := bodyProgress
 	noProgressCount := 0 // How many time the progress was printed without actual progress
-Loop:
-	for !stopped {
-		// TODO: this is incorrect use
-		if req == nil {
-			start := time.Now()
-			currentTime := uint64(time.Now().Unix())
-			req, blockNum, err = cfg.bd.RequestMoreBodies(tx, cfg.blockReader, blockNum, currentTime, cfg.blockPropagator)
+	var totalDelivered uint64 = 0
+
+	// create a temporary bucket to fire the bodies into as we start to collect them
+	// this will allow us to restart the bodies stage and not request bodies we already have
+	// once the bodies stage is complete this bucket is dropped
+	if !useExternalTx {
+		err = tx.CreateBucket("BodiesStage")
```

Comment:
What would happen if the bucket exists?

Reply:
It's a no-op if the bucket already exists, I asked a few questions on Discord to check at the time